src/store/mutations.ts

```diff
@@ -148,16 +149,23 @@ export default {
    * Delete the step of index `index` in pipeline.
    */
   deleteStep(state: VQBState, { index }: { index: number }) {
-    state.pipelines[state.currentPipelineName].splice(index, 1);
+    const pipeline = currentPipeline(state);
+    if (!pipeline) {
+      return;
+    }
+    pipeline.splice(index, 1);
     state.selectedStepIndex = index - 1;
   },
   /**
    * change current selected domain and reset pipeline accordingly.
    */
   setCurrentDomain(state: VQBState, { currentDomain }: Pick<VQBState, 'currentDomain'>) {
+    const pipeline = currentPipeline(state);
+    if (state.currentPipelineName == null || !pipeline) {
```

Comment:
It seems that `currentPipelineName` can't be `null` from a type point of view so I guess 

```suggestion
    if (state.currentPipelineName === undefined || pipeline === undefined) {
```